### DISCUSSION TITLE
Error when trying to change the livewire/update endpoint to a custom endpoint

### DISCUSSION THREAD

**Discussion by geordym** [livewire/livewire#8233] [7. Bugs (archived)]
### Livewire version

v.3.4.9

### Laravel version

v10.10

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Open  routes/web.php
2. Include that lines

```
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/custom/livewire/update', $handle);
});
```

3. Use **php artisan route:cache**

### Repository or Wirebox URL

https://github.com/livewire/livewire/releases/tag/v3.4.9

### Additional information

When i'm trying to change the endpoint of livewire/update, because i have my project in an subdir localhost/project/public/ but livewire is searching the update endpoint in localhost/livewire/update, when i read the docs of 3.0 to solve that, i give that error at the moment of use php artisan route:cache 
 
  Unable to prepare route [custom/livewire/update] for serialization. Another route has already been assigned name [livewire.update].

  at vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:247
    243▕             $route->name($this->generateRouteName());
    244▕
    245▕             $this->add($route);
    246▕         } elseif (! is_null($symfonyRoutes->get($name))) {
  ➜ 247▕             throw new LogicException("Unable to prepare route [{$route->uri}] for serialization. Another route has already been assigned name [{$name}].");

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@geordym try giving your custom route a name as it would likely be colliding with the inbuilt one.

```php
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/custom/livewire/update', $handle)->name('custom-update');
});
```

**Reply by geordym** [reply]
I try that and the problem continue

**Reply by jerry-ye** [reply]
have you tried to name your custom route path? **->name('custom-update');**

**Reply by awesomevilo-abad** [reply]
Hi, would like to asked if this is solve?, adding name to the route will solve the caching issue but will trigger the original error  (livewire.js is not found) in console.

**Reply by ikp786** [reply]
I am facing same issue if you find any solution then please let me know thanks

**Reply by ikp786** [reply]
I fix this issue using I change in web.php file

// Define the base URL for Livewire update route
**$livewireUpdateRoute = env('LIVEWIRE_UPDATE_ROUTE');**

// Set the Livewire update route dynamically
**Livewire::setUpdateRoute(function ($handle) use ($livewireUpdateRoute) {
    return Route::post($livewireUpdateRoute, $handle);
});**

.env file

**LIVEWIRE_UPDATE_ROUTE="/stucard/livewire/update"**

this is working fine but I don't know this is better way or not

**Comment by mcafeekj** [comment]
I am having the same issue.  Unfortunately, the workaround did not work for me.  Giving the route a unique name completely broke my application.  I have just resigned myself to not being able to use route:clear for now.

**Comment by Polfo** [comment]
For me giving the route a custom name with `->name('custom-livewire.update');` prevents an error on `artisan route:cache` 
```
Livewire::setUpdateRoute(function ($handle) {
     return Route::post(config('app.asset_url').'/livewire/update', $handle)->name('custom-livewire.update');
});
```
Based on the code below I think the name should be ending with `livewire.update`. But this may be a temporary solution. See `vendor/livewire/livewire/src/Mechanisms/HandleRequests/HandleRequests.php`

    function isLivewireRoute()
    {
        // @todo: Rename this back to `isLivewireRequest` once the need for it in tests has been fixed.
        $route = request()->route();

        if (! $route) return false;

        /*
         * Check to see if route name ends with `livewire.update`, as if
         * a custom update route is used and they add a name, then when
         * we call `->name('livewire.update')` on the route it will
         * suffix the existing name with `livewire.update`.
         */
        return $route->named('*livewire.update');
    }

**Comment by jkleintele** [comment]
Did anyone ever found a workaround to having to set
`Livewire::setUpdateRoute`

inside the web.php routes file?

As others has said when you run console commands you get this error:

`Unable to prepare route [nexus/livewire/update] for serialization. Another route has already been assigned name [livewire.update].`

**Reply by MichalMeliska** [reply]
still not solved
adding custom name to route fixed the caching issue but in route list, there is still default update endpoint together with new one

POST       livewire/update ........................................................................... livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate  
POST       mm/public/livewire/update .......................................................... custom-livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate

**Reply by joshhanley** [reply]
@MichalMeliska that's because the `setUpdateRoute()` adds a new route with your definition, but it won't remove the old one. The old one just gets left unused. Adding a custom name to your new route is how you stop the name conflict.

**Comment by Tapuwa-Dhliwayo** [comment]
We have found a work around that resolves this issue but it requires changing the getUpdateUri in the vendor/livewire/livewire/src/LivewireManager.php file to the following:
function getUpdateUri()
    {
        return rtrim(request()->getBaseUrl(), '/') .  app(HandleRequests::class)->getUpdateUri();
    }
    
    Hopefully this gets looked at and addressed
